### PR TITLE
Cover non-empty-array in `array_combine`

### PR DESCRIPTION
**Pull request by staabm**
Refs https://github.com/phpstan/phpstan/issues/5327

### CODE REVIEW

File: tests/PHPStan/Analyser/LegacyNodeScopeResolverTest.php [hunk 1]
```diff
@@ -5009,11 +5009,11 @@ public function dataArrayFunctions(): array
 				'array_combine([\'a\', \'b\', \'c\'], [\'d\', \'e\', \'f\'])',
 			],
 			[
-				PHP_VERSION_ID < 80000 ? 'array<1|2|3, mixed>|false' : 'array<1|2|3, mixed>',
+				PHP_VERSION_ID < 80000 ? 'array<1|2|3, mixed>|false' : 'array<1|2|3, mixed>&nonEmpty',
```
Comment (staabm) [inline tests/PHPStan/Analyser/LegacyNodeScopeResolverTest.php:5012]:
```suggestion
				PHP_VERSION_ID < 80000 ? 'array<1|2|3, mixed>|false' : 'array<1|2|3, mixed>',
```

File: tests/PHPStan/Analyser/LegacyNodeScopeResolverTest.php [hunk 2]
```diff
@@ -5009,11 +5009,11 @@ public function dataArrayFunctions(): array
...
 				'array_combine([1, 2, 3], $array)',
 			],
 			[
-				PHP_VERSION_ID < 80000 ? 'array<1|2|3>|false' : 'array<1|2|3>',
+				PHP_VERSION_ID < 80000 ? 'array<1|2|3>|false' : 'array<1|2|3>&nonEmpty',
```
Comment (staabm) [inline tests/PHPStan/Analyser/LegacyNodeScopeResolverTest.php:5016]:
```suggestion
				PHP_VERSION_ID < 80000 ? 'array<1|2|3>|false' : 'array<1|2|3>',
```

**Comment by staabm** [comment]
ready to merge

**Comment by ondrejmirtes** [comment]
Thank you!

**Pull request merged**